lib/ZwaveClient.js

```diff
@@ -110,183 +46,375 @@ function ZwaveClient (config, socket) {
+}
+
+function onNodeSleep (
+  zwaveNode,
+  oldStatus
+) {
+  debug(
+    `Node ${zwaveNode.id} is ${
+      oldStatus === NodeStatus.Unknown ? '' : 'now '
+    }asleep`
+  )
+  onNodeStatus.call(this, zwaveNode)
+}
+
+function onNodeAlive (
+  zwaveNode,
+  oldStatus
+) {
+  onNodeStatus.call(this, zwaveNode)
+  if (oldStatus === NodeStatus.Dead) {
+    debug(`Node ${zwaveNode.id}: has returned from the dead`)
+  } else {
+    debug(`Node ${zwaveNode.id} is alive`)
   }
 }
 
-function driverFailed () {
-  this.error = 'Driver failed'
-  this.status = ZWAVE_STATUS[5]
-  debug('Driver failed', this.ozwConfig)
+function onNodeDead (
+  zwaveNode,
+  oldStatus
+) {
+  onNodeStatus.call(this, zwaveNode)
+  debug(
+    `Node ${zwaveNode.id} is ${
+      oldStatus === NodeStatus.Unknown ? '' : 'now '
+    }dead`
+  )
+}
+
+// function onNodeValueAdded (
+//   zwaveNode,
+//   args
+// ) {
+//   debug(
+//     `Node ${zwaveNode.id}: value added: ${getValueID(args)} => ${String(
+//       args.newValue
+//     )}`
+//   )
+// }
+
+function onNodeValueUpdated (
+  zwaveNode,
+  args
+) {
+  updateValue.call(this, zwaveNode, args)
+  debug(
+    `Node ${zwaveNode.id}: value updated: ${getValueID(args)} ${args.prevValue} => ${String(
+      args.newValue
+    )}`
+  )
+}
+
+function onNodeValueRemoved (
+  zwaveNode,
+  args
+) {
+  removeValue.call(this, zwaveNode, args)
+  debug(`Node ${zwaveNode.id}: value removed: ${args}`)
+}
+
+function onNodeMetadataUpdated (
+  zwaveNode,
+  args
+) {
+  updateValueMetadata.call(this, zwaveNode, args, args.metadata)
+  debug(`Node ${zwaveNode.id}: metadata updated: ${getValueID(args)}`)
 }
 
-function connected (version) {
-  this.ozwConfig.version = version
-  debug('Zwave connected, Openzwave version:', version)
-  this.status = ZWAVE_STATUS[1]
+function onNodeFirmwareUpdateProgress (
+  zwaveNode,
+  sentFragments,
+  totalFragments
+) {
+  debug(`Node ${zwaveNode.id} firmware update IN PROGRESS: ${sentFragments}/${totalFragments}`)
+}
 
-  this.emitEvent('CONNECTED', this.ozwConfig)
+// https://github.com/zwave-js/node-zwave-js/blob/cb35157da5e95f970447a67cbb2792e364b9d1e1/packages/zwave-js/src/lib/commandclass/FirmwareUpdateMetaDataCC.ts#L59
+function onNodeFirmwareUpdateFinished (
+  zwaveNode,
+  status,
+  waitTime
+) {
+  debug(`Node ${zwaveNode.id} firmware update FINISHED: Status ${status}, Time: ${waitTime}`)
 }
 
-function nodeRemoved (nodeid) {
+// ------- NODE METHODS -------------
+
+function bindNodeEvents (node) {
+  // add a node to our nodes array
+  addNode.call(this, node)
+
+  // https://zwave-js.github.io/node-zwave-js/#/api/node?id=zwavenode-events
+  node.on('ready', onNodeReady.bind(this))
+    .on('interview completed', onNodeInterviewCompleted.bind(this))
+    .on('wake up', onNodeWakeUp.bind(this))
+    .on('sleep', onNodeSleep.bind(this))
+    .on('alive', onNodeAlive.bind(this))
+    .on('dead', onNodeDead.bind(this))
+  // .on('value added', onNodeValueAdded.bind(this))
```

Comment:
I would attach that handler or you will miss all values during the initial interview of a node.

Reply:
@AlCalzone Nope, I removed it on purpose, I decided to add them on node ready event